tests/kernel/mem_protect/mem_protect/src/inherit.c

```diff
@@ -78,11 +79,11 @@ void test_thread_1_for_SU(void *p1, void *p2, void *p3)
  */
 void test_permission_inheritance(void *p1, void *p2, void *p3)
 {
-
 	k_mem_domain_init(&inherit_mem_domain,
-			  1,
+			  ARRAY_SIZE(inherit_memory_partition_array),
 			  inherit_memory_partition_array);
 
+	k_mem_domain_remove_thread(k_current_get());
```

Comment:
Why is the thread removed, now?

Reply:
I see that this is common practice now, pls, explain briefly :)

Reply:
Before this change, the thread was not a member of any domain. However it is now in ztest's domain. You have to remove from existing domain before joining a new one else the memory domain code asserts.